components/workflow-mgt/org.wso2.carbon.identity.workflow.mgt/src/main/java/org/wso2/carbon/identity/workflow/mgt/listener/WorkflowListener.java

```diff
@@ -650,4 +650,21 @@ void doPostListEntityNames(String wfOperationType, String wfStatus, String entit
      * @return
      */
     int getOrderId();
+
+    /**
+    * Trigger before retrieving a workflow request.
+    *
+    * @param requestId ID of the workflow request to retrieve
+    * @throws WorkflowException
+    */
+    void doPreGetWorkflowRequest(String requestId) throws WorkflowException;
```

Comment:
When we add new methods to the interface we follow the default modifier.
Reference: https://www.baeldung.com/java-static-default-methods